core/services/relay/evm/chain_writer.go

```diff
@@ -155,8 +166,42 @@ func (w *chainWriter) GetTransactionStatus(ctx context.Context, transactionID uu
 	return commontypes.Unknown, fmt.Errorf("not implemented")
 }
 
+// GetFeeComponents the execution and data availability (L1Oracle) fees for the chain.
+// Dynamic fees (introduced in EIP-1559) include a fee cap and a tip cap. If the dyanmic fee is not available,
+// (if the chain doesn't support dynamic TXs) the legacy GasPrice is used.
 func (w *chainWriter) GetFeeComponents(ctx context.Context) (*commontypes.ChainFeeComponents, error) {
-	return nil, fmt.Errorf("not implemented")
+	if w.ge == nil {
+		return nil, fmt.Errorf("not implemented")
+	}
+
+	fee, _, err := w.ge.GetFee(ctx, nil, 0, assets.NewWei(w.maxGasPrice))
+	if err != nil {
+		return nil, err
+	}
+	// Use legacy if no dynamic is available.
+	gasPrice := fee.Legacy.ToInt()
+	if fee.DynamicFeeCap != nil {
+		gasPrice = fee.DynamicFeeCap.ToInt()
+	}
+	if gasPrice == nil {
+		return nil, fmt.Errorf("Dynamic fee and legacy gas price missing %+v", fee)
```

Comment:
Error messages are meant to be lower case:
```suggestion
		return nil, fmt.Errorf("dynamic fee and legacy gas price missing %+v", fee)
```
Linter is supposed to catch this one... :thinking: